book/src/linking_native_functions.md

```diff
@@ -0,0 +1,96 @@
+# Linking native functions into the JVM
+
+Using the [`#[java_function]`](./java_function.md) decorator you can write Rust implementations for Java native functions. To get the JVM to invoke these methods, it has to know how to find them. The way you do this depends on whether you have the "top-level" program is written in Rust or in Java.
+
+## Rust program that creates a JVM
+
+If your Rust program is launching the JVM, then you can configure that JVM to link to your native method definitions through methods on the JVM builder.
+
+```rust
+use duchess::prelude::*; // 👈 You'll need this.
+
+#[java_function(...)]
+fn foo(...) { }
+
+fn main() -> duchess::GlobalResult<()> {
+    Jvm::builder()
+        .link(foo::java_fn()) // 👈 Note the `::java_fn()`.
+        .try_launch()?;
+}
+```
+
+**How it works.** The call `foo::java_fn()` returns a `duchess::JavaFunction` struct. The `java_fn` method is defined in the duchess `JavaFn` trait; that trait is implemented on a struct type `foo` that is created by the `#[java_function]` decorator. This trait is in the duchess prelude, which is why you need to `use duchess::prelude::*`.
+
+### Java function suites
+
+Invoking the link method for every java functon you wish to implement is tedious and error-prone. If you have java functions spread across crates and modules, it also presents a maintenance hazard, since each time you add a new `#[java_function]` you would also have to remember to add it to the Jvm builder invocation, which is likely located in some other part of the code.
+
+To avoid this, you can create **suites** of java functions. The idea is that the `link` method accepts both individual `JavaFunction` structs but also `Vec<JavaFunction>` suites. You can then write a function in your module that returns a `Vec<JavaFunction>` with all the java functions defined locally:
+
+```rust
+use duchess::prelude::*;
```

Comment:
I think `JavaFunctions` is not in the prelude, because I tried defining a `java_functions()` in the `native_greetings.rs` test and I get the following:
```
error[E0412]: cannot find type `JavaFunction` in this scope
  --> tests/ui/native_greeting.rs:24:28
   |
24 | fn java_functions() -> Vec<JavaFunction> {
   |                            ^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
3  | use duchess::plumbing::JavaFunction;
   |
```

Reply:
D'oh. Fixed. It's not in the *prelude* but it is in the crate outside of `plumbing` now.

Side note but I wonder if we can find a better way to deal with `plumbing`. It'd be cool if you could annotate something so the compiler wouldn't offer it in suggestions; or perhaps if the compiler would just avoid "doc-hidden" suggestions.